site/src/components/WorkspaceActions/WorkspaceActions.tsx

```diff
@@ -83,26 +90,62 @@ export const WorkspaceActions: FC<WorkspaceActionsProps> = ({
     ),
   }
 
-  // memoize so this isn't recalculated every time we fetch the workspace
-  const [primaryAction, ...secondaryActions] = useMemo(
-    () =>
-      isUpdating
-        ? [ButtonTypesEnum.updating, ...actions]
-        : canBeUpdated
-        ? [ButtonTypesEnum.update, ...actions]
-        : actions,
-    [actions, canBeUpdated, isUpdating],
-  )
+  // Returns a function that will execute the action and close the menu
+  const onMenuItemClick = (actionFn: () => void) => () => {
+    setIsMenuOpen(false)
+    actionFn()
+  }
 
   return (
-    <DropdownButton
-      primaryAction={buttonMapping[primaryAction]}
-      canCancel={canCancel}
-      handleCancel={handleCancel}
-      secondaryActions={secondaryActions.map((action) => ({
-        action,
-        button: buttonMapping[action],
-      }))}
-    />
+    <div className={styles.actions} data-testid="workspace-actions">
+      {canBeUpdated &&
+        (isUpdating
+          ? buttonMapping[ButtonTypesEnum.updating]
+          : buttonMapping[ButtonTypesEnum.update])}
+      {actionsByStatus.map((action) => buttonMapping[action])}
```

Comment:
I think we need a key here so we don't get a console error:
![Screenshot 2023-04-14 at 10 22 20 AM](https://user-images.githubusercontent.com/19142439/232071364-cd548e4a-c6af-47d6-9afd-4b7e4a7c5b31.png)

Reply:
Good catch!